hack/ensure-azcli.sh

```diff
@@ -35,6 +35,6 @@ if [[ -z "$(command -v az)" ]]; then
     # Use --auth-mode "login" in az storage commands to use RBAC permissions of login identity. This is a well known ENV variable the Azure cli
     export AZURE_STORAGE_AUTH_MODE="login"
   else
-    az login --service-principal -u "${AZURE_CLIENT_ID}" -p "${AZURE_CLIENT_SECRET}" --tenant "${AZURE_TENANT_ID}" > /dev/null
+    exit 1
```

Comment:
```suggestion
    echo "federated token not set. Exiting"
    exit 1
```

Having a useful message before exit should help in triaging.

Reply:
good suggestion!